File: Basics/While-Loops/Exercise/Resources/exam_preparation.py
max_bad_appraisals = int(input())
problem_name = input()
last_problem = ""
appraisals_sum = 0
total_problems = 0
bad_appraisal_count = 0
is_enough = False

while problem_name != "Enough":
    current_appraisal = int(input())
    total_problems += 1
    appraisals_sum += current_appraisal
    last_problem = problem_name

    if current_appraisal <= 4:
        bad_appraisal_count += 1

    if bad_appraisal_count >= max_bad_appraisals:
        is_enough = True
        break

    problem_name = input()

if is_enough:
    print(f"You need a break, {max_bad_appraisals} poor grades.")
else:
    print(f"Average score: {appraisals_sum / total_problems:.2f}\n"
          f"Number of problems: {total_problems}\n"
          f"Last problem: {last_problem}")

File: JavaScript/Basics/0.5-While-Loops/Exercises/exam-preparation.js
function examPreparation(input) {
    let maxBadAppraisals = Number(input.shift()); let problemsCount = 0;
    let i = 0; let averageScore = 0; let badAppraisals = 0; let problem = "";
    while (input[i] != "Enough") {
        problem = input[i]; let appraisal = Number(input[i + 1]);
        if (appraisal <= 4) {
            badAppraisals += 1;
            if (badAppraisals ==  maxBadAppraisals) {
                return `You need a break, ${badAppraisals} poor grades.`;
            }
        } 
        i += 2; problemsCount++; averageScore += appraisal;
    }
    averageScore = averageScore / problemsCount
    return `Average score: ${averageScore.toFixed(2)}\nNumber of problems: ${problemsCount}\nLast problem: ${problem}`;
}


console.log(examPreparation(["3",
"Money",
"6",
"Story",
"4",
"Spring Time",
"5",
"Bus",
"6",
"Enough"]));
console.log(examPreparation(["2",
"Income",
"3",
"Game Info",
"6",
"Best Player",
"4"]));